### PR TITLE
Fix android build

### PR DESCRIPTION
**Pull request by cajus**
closes https://github.com/bincrafters/community/issues/710 
These changes helped me when building Qt for Android. Note that not all requirements seem to be there in *bincrafters* - i.e. harfbuzz needs to be fixed. And an android SDK package is needed as a dependency in the profile.

I've opened the PR for sharing and further discussion as a follow up of https://github.com/bincrafters/community/issues/710.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -534,9 +547,10 @@ def _getenvpath(var):
 
         value = _getenvpath('CXX')
         if value:
-            args += ['QMAKE_CXX="' + value + '"',
-                     'QMAKE_LINK="' + value + '"',
-                     'QMAKE_LINK_SHLIB="' + value + '"']
+            args += ['QMAKE_CXX="' + value + '"']
```
Comment (ericLemanissier) [inline conanfile.py]: Is this change still needed with the `-nostdlib++` patch below ?

Comment (cajus) [reply conanfile.py]: You're right. That's a left over.

File: conanfile.py [hunk 2]
```diff
@@ -299,6 +301,9 @@ def source(self):
             self.run("wget -qO- %s.tar.xz | tar -xJ " % url)
         shutil.move("qt-everywhere-src-%s" % self.version, "qt5")
 
+        for patch in glob.glob("patches/*.diff"):
```
Comment (SSE4) [inline conanfile.py]: please rename patch to start with `0001-` and add `sorted(glob.glob(...))` here.
it's important for the future that patches apply in the right and consistent order.

**Comment by ericLemanissier** [comment]
@cajus thanks a lot for the PR. Did you submit the patch to Qt ?

**Comment by cajus** [comment]
Indirectly. I've sent it to a Qt guy I know, but I didn't open a change there. The question is: how backwards compatible is the contribution for the project in general.

**Comment by ericLemanissier** [comment]
This recipe is now in the Conan Center Index.

Please migrate to the recipe version there.
If this is still an issue, please create a new pull request in the Conan Center Index repository.

Thanks!

https://conan.io/center/qt
